R/create_epa.R

```diff
@@ -159,9 +167,31 @@ create_epa <- function(clean_pbp_dat,
   # For turnover and punt plays make sure the ep_after is negative
   # because of poor ESPN data quality,
   # some drives end on 3rd down and we have those listed as turnovers
-  turnover_plays = which(pred_df$turnover == 1 & !kickoff_ind & pred_df$play_type %in% turnover_play_type)
+  turnover_plays = which(pred_df$turnover == 1 & !kickoff_ind & (pred_df$play_type %in% turnover_play_type))
```

Comment:
Where do we set pred_df$turnover? 

Do we do that using the same `turnover_play_type` object? I can't seem to find it.

Reply:
I would also separate out the boolean into variables. Would be cleaner to read.

Reply:
We define it in `prep_df_epa_after`. So ignore the question

Reply:
So instead of doing this here. I think we should properly define this. 

```  turnover_ind = dat$play_type %in% turnover_play_type
  dat$turnover = 0
  #define turnover on downs
  downs_turnover = (dat$play_type %in% normalplay & dat$down == 4)
  # data is ordered
  new_offense = !(dat$offense_play == lead(dat$offense_play,order_by = dat$id_play))
  scoring_plays = dat$play_type %in% score
  # end of half check as well
  end_of_half_plays = !(dat$half == lead(dat$half,order_by = dat$id_play))
  # is specifically defined as a turnover
  turnover_play_check = dat$play_type %in% turnover_vec
  # turnovers only occur on actual change of offense
  # but not scoring plays
  # and not at the end of half.
  # Turnovers now capture downs, when there is a change of offense after a fourth down normal play.
  t_ind = (turnover_ind | (new_offense)) & !scoring_plays & !end_of_half_plays & (turnover_play_check | downs_turnover)

  dat$turnover[t_ind] <- 1
```

It is defined in `prep_df_epa_after` and I feel like we are duplicating the work in this line.